### DISCUSSION TITLE
Review Caching

### DISCUSSION THREAD

**Discussion by scottcwilson** [zencart/zencart#5788] [Ideas]
Note: Original title was "should store manager provide a cache clear button?"  I realized I needed to expand the scope of this inquiry. 

Work Items: 
- [x] verify behavior of cache in 1.5.7
- [ ] provide documentation on how to use this.
- [ ] link to new cache documentation from performance documentation.
- [X] provide a clear function in admin - DONE - performed already on every admin page load by `sql_cache_flush_cache`.
- [X] document disabling the products viewed counter - DONE see https://docs.zen-cart.com/user/performance/speed_up_site/#disable-products-viewed
- [X] add cache logic to 1.58
- [ ] get the cache PR merged in.

**Comment by drbyte** [comment]
When would this be used? ie: what's the business problem you've run into that this would solve?

The cache is already cleared every time an Admin configuration_value is changed.

**Comment by scottcwilson** [comment]
> The cache is already cleared every time an Admin configuration_value is changed.

I'm not seeing this - where is this logic? 

I'm thinking about the situation where you make a bunch of product changes and you don't want to show old data.

**Comment by drbyte** [comment]
> The cache is already cleared every time an Admin configuration_value is changed.

Correction: It's cleared on every Admin page hit:

https://github.com/zencart/zencart/blob/b21f28f541b8bab6d299de9c8417f49f6cab34af/admin/includes/init_includes/init_database.php#L51-L52

**Comment by scottcwilson** [comment]
Sweet!   

I wonder if the sql_cache_flush_cache 'memory' case should clear `$this->cache_array` just in case someone implements 'memory' fully in the future.

**Comment by drbyte** [comment]
> I wonder if the sql_cache_flush_cache 'memory' case should clear `$this->cache_array` just in case someone implements 'memory' fully in the future.

Correct observation; however I'd leave that change for when it becomes context-relevant.

**Comment by scottcwilson** [comment]
oh agreed,  I might just add a comment or something if I ever happen to be in there.  2/5 items done - whoo!

**Comment by scottcwilson** [comment]
I have these changes in test locally and they look good.  I'll push a branch in a few weeks when the 1.5.7 activity settles down.

**Reply by drbyte** [reply]
Did this branch get pushed?

**Comment by scottcwilson** [comment]
In test now.  My intention is to change the behavior described in this post https://github.com/zencart/zencart/issues/3403#issuecomment-623187248 so that a storeowner can process orders in admin without clearing the cache.
